File: ui-automation-java/src/test/java/com/orangehrm/pages/SideMenu.java
package com.orangehrm.pages;

import org.openqa.selenium.By;

/**
 * Left side menu navigation for OrangeHRM.
 */
public class SideMenu extends BasePage {

    private By menuItem(String name) {
        return By.xpath("//span[contains(@class,'oxd-main-menu-item--name') and normalize-space(.)='" + name + "']/ancestor::a");
    }

    public void openDashboard() {
        safeClick(menuItem("Dashboard"));
    }

    public void openPim() {
        safeClick(menuItem("PIM"));
    }

    public void openLeave() {
        safeClick(menuItem("Leave"));
    }

    public void openRecruitment() {
        safeClick(menuItem("Recruitment"));
    }

    public void openTime() {
        safeClick(menuItem("Time"));
    }

    public void openAdmin() {
        safeClick(menuItem("Admin"));
    }
}

File: ui-automation-csharp/OrangeHrm.UiTests/Pages/SideMenu.cs
using OpenQA.Selenium;

namespace OrangeHrm.UiTests.Pages;

public class SideMenu : BasePage
{
    private By MenuItem(string name) =>
        By.XPath($"//span[contains(@class,'oxd-main-menu-item--name') and normalize-space(.)='{name}']/ancestor::a");

    public void OpenDashboard() => Click(MenuItem("Dashboard"));
    public void OpenPim() => Click(MenuItem("PIM"));
    public void OpenLeave() => Click(MenuItem("Leave"));
    public void OpenRecruitment() => Click(MenuItem("Recruitment"));
    public void OpenTime() => Click(MenuItem("Time"));
    public void OpenAdmin() => Click(MenuItem("Admin"));
}
